src/hooks/useSwapDerivedOutputs.ts

```diff
@@ -316,237 +300,259 @@ export default function useSwapDerivedOutputs(chainId: number, type: string) {
     genericAssets[outputCurrency?.mainnet_address || outputCurrency?.address]
       ?.price?.value;
 
-  const { accountAddress } = useAccountSettings();
-
-  const resetSwapInputs = () => {
+  const resetSwapInputs = useCallback(() => {
     derivedValues[SwapModalField.input] = null;
     derivedValues[SwapModalField.output] = null;
     derivedValues[SwapModalField.native] = null;
     displayValues[DisplayValue.input] = null;
     displayValues[DisplayValue.output] = null;
     displayValues[DisplayValue.native] = null;
-    setResult({ derivedValues, displayValues, tradeDetails: null });
-    setLoading(false);
-    setInsufficientLiquidity(false);
-  };
+    dispatch(
+      updateSwapQuote({
+        derivedValues,
+        displayValues,
+        tradeDetails: null,
+      })
+    );
+    return {
+      insufficientLiquidity: false,
+      result: { derivedValues, displayValues, tradeDetails: null },
+    };
+  }, [dispatch]);
+
+  const getTradeDetails = useCallback(async () => {
+    let tradeDetails = null;
+
+    if (independentValue === '0.') {
+      switch (independentField) {
+        case SwapModalField.input:
+          displayValues[DisplayValue.input] = independentValue;
+          break;
+        case SwapModalField.output:
+          displayValues[DisplayValue.output] = independentValue;
+          break;
+        case SwapModalField.native:
+          displayValues[DisplayValue.native] = independentValue;
+          break;
+      }
 
-  useEffect(() => {
-    const getTradeDetails = async () => {
-      let tradeDetails = null;
-
-      if (independentValue === '0.') {
-        switch (independentField) {
-          case SwapModalField.input:
-            displayValues[DisplayValue.input] = independentValue;
-            break;
-          case SwapModalField.output:
-            displayValues[DisplayValue.output] = independentValue;
-            break;
-          case SwapModalField.native:
-            displayValues[DisplayValue.native] = independentValue;
-            break;
-        }
-        setResult({
+      return {
+        insufficientLiquidity: false,
+        result: {
           derivedValues,
           displayValues,
-          tradeDetails,
-        });
-        return;
-      }
+        },
+      };
+    }
 
-      if (
-        (isZero(independentValue) && independentValue.length === 1) ||
-        !independentValue
-      ) {
-        resetSwapInputs();
-        return;
-      }
+    if (!independentValue) {
+      return resetSwapInputs();
+    }
+
+    const inputToken = inputCurrency;
+    const outputToken = outputCurrency;
```

Comment:
I don't see an issue with this but curious why you are cloning these values. I don't see anything in getOutputAmount that requires you to do so.